sdk/search/azure-search-documents-serializer/src/main/java/com.azure.search.documents.serializer/SearchSerializerProvider.java

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.serializer;
+
+/**
+ * Interface to be implemented by an azure-core plugin that wishes to provide a {@link SearchSerializer} implementation.
+ */
+public interface SearchSerializerProvider {
```

Comment:
I believe we would want to stay away from the Search serializer modules having a service provider interface, I feel instead they should wrap the Azure Core variant otherwise it'll be confusing to know what serialization implementation to use.

Another option would be the Search variants implement and provide the Core interface instead of their own.

Reply:
I have posted in team channel. Search needs to have type as parameter otherwise we are not able to do map serialization.
I am ok to move to azure-core-experiment module API if it provides the corresponding API